packages/salesforce-adapter/src/client/client.ts

```diff
@@ -281,27 +281,36 @@ export default class SalesforceClient {
     }
   )
 
-  private static logDecorator = decorators.wrapMethodWith(
-    // eslint-disable-next-line prefer-arrow-callback
-    async function logFailure(
-      this: SalesforceClient,
-      { call, name, args }: decorators.OriginalCall,
-    ): Promise<unknown> {
-      const desc = `client.${name}(${args.map(arg => _.get(arg, 'fullName', arg))
-        .filter(arg => typeof arg === 'string').join(', ')})`
-      try {
-        return await log.time(call, desc)
-      } catch (e) {
-        log.error('failed to run SFDC client call %s: %s', desc, e.message)
-        throw e
+  private static logDecorator = (keys?: string[]): decorators.InstanceMethodDecorator =>
+    decorators.wrapMethodWith(
+      // eslint-disable-next-line prefer-arrow-callback
+      async function logFailure(
+        this: SalesforceClient,
+        { call, name, args }: decorators.OriginalCall,
+      ): Promise<unknown> {
+        const printableArgs = args
+          .map(arg => {
+            const keysValues = (keys ?? [])
+              .map(key => _.get(arg, key))
+              .filter(val => !_.isUndefined(val))
```

Comment:
instead of a double negative
```suggestion
              .filter(values.isDefined)
```

Reply:
actually, thinking about it again, do we need to filter to make sure we only get strings here like we do for the other args?